### PR TITLE
Feature 5730 profile page updates

### LINKED ISSUE #5730: ✨ Applicant Dashboard - Updates to Profile Page
## ✨ Feature

Updates to the profile based on the new Dashboard page

## 🕵️ Details

The dashboard causes a few changes that required to the existing profile page to help with cohesiveness and clarity:
- Remove the "status" section and ensure its removal doesn't impact candidate and admin functionality
- Remove the "My skills and experience" section
- Delete unused My Hiring Pools section in the code
- Update the "On this page" sidebar to reflect section completion, matching the hero area of the dashboard

This work should all be behind the feature flag.

## 📸 Screenshot
<details>
  <summary><strong>📸  Screenshot: full updated profile</strong></summary>
  
</details>

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

### Changes to the profile
- [ ] Feature flag is in place to keep any of these changes from appearing on the profile until we want them to
- [ ] The profile no longer contains a "My status" section
- [ ] The applicant can still successfully apply to opportunities despite "My status" being removed
- [ ] The "My skills and experience" section has been removed and users can now view their resume information via the dashboard hero area (which links to the Skills and experience page)
- [ ] The "On this page" sidebar showcases completion status of each section and matches the look and feel of the dashboard profile summary in the hero area

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #5730 
## 👋 Introduction

This branch updates the profile page with status icons on the sidebar and removes the status and experiences sections. 

## 🕵️ Details

The "profile complete" toast was also disabled since the section it referred to was removed.

The issue scope was expanded to removing the "available for opportunities" and "job looking status" query scopes in [this later discussion](https://talent-cloud.slack.com/archives/C0259G6KXJ6/p1678387944415379) on Slack.  This required a bunch of test changes, too.

## 🧪 Testing

1. Log into the app as an applicant and review the profile page.
2. Manually make the profile complete and incomplete to observe the sidebar and toast changes. 
3. Flip the FEATURE_APPLICANT_DASHBOARD flag to false and confirm the page looks and behaves like before.
4. Search works OK (other than no longer filtering unavailable and inactive candidates).

## 📸 Screenshot

image

### CODE REVIEW

**Comment by esizer** [comment]
Holy moly this is impressive! One tiny nit pick, can `apps/web/src/component/infoItem` directory use `PascalCase` 😅

**Comment by vd1992** [comment]
will need to update the users table as well at some point, spun off to its own issue maybe (not sure if an issue already exists) 

- remove the deprecated field from the table

image

- update the filter (setting it doesn't break anything, it just does nothing now so should be removed in lockstep with this stuff)

image

**Comment by petertgiles** [comment]
> will need to update the users table as well at some point, spun off to its own issue maybe (not sure if an issue already exists)

Good call.  I've added #5937 to the board.

File: apps/web/src/components/InfoItem/BaseInfoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          data-h2-margin-top="base(x.15)"
+          data-h2-transition="base(color .2s ease)"
+          {...iconColorMap[iconColor]}
+        />
+      )}
+
+      <div>
+        {titleHref ? (
+          <Link href={titleHref} {...{ ...textColorMap[titleColor] }}>
+            {title}
```
Comment (esizer) [inline apps/web/src/components/InfoItem/BaseInfoItem.tsx:99]: You likely don't want to read this but, can we indicate status to assistive technology in some way? Like prending hidden text?

```tsx
<span data-h2-visually-hidden="base(hidden)">Incomplete section, </span>{title}
 
<span data-h2-visually-hidden="base(hidden)">Completed section, </span>{title}
```

**Review by esizer — CHANGES_REQUESTED** [review]
This is looking great! One sorta small suggestion for accessibility 😄

File: apps/web/src/components/InfoItem/BaseInfoItem.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/components/InfoItem/BaseInfoItem.tsx]: Nice! Sounds good to me 😄

**Review by esizer — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
...
     "check-intl:web": "turbo run check-intl --filter=@gc-digital-talent/i18n -- --dir $INIT_CWD --en ./apps/web/src/lang/en.json --fr ./apps/web/src/lang/fr.json --rm-orphaned --output-untranslated ./apps/web/src/lang/untranslated.json --whitelist ./apps/web/src/lang/whitelist.yml",
-    "check-intl-merge:i18n": "turbo run check-intl --filter=@gc-digital-talent/i18n -- --dir $INIT_CWD --en ./packages/i18n/src/lang/en.json --fr ./packages/i18n/src/lang/fr.json --rm-orphaned --output-untranslated ./packages/i18n/src/lang/untranslated.json --whitelist ./packages/i18n/src/lang/whitelist.yml --merge-fr ./packages/i81n/src/lang/newTranslations.json",
+    "check-intl-merge:i18n": "turbo run check-intl --filter=@gc-digital-talent/i18n -- --dir $INIT_CWD --en ./packages/i18n/src/lang/en.json --fr ./packages/i18n/src/lang/fr.json --rm-orphaned --output-untranslated ./packages/i18n/src/lang/untranslated.json --whitelist ./packages/i18n/src/lang/whitelist.yml --merge-fr ./packages/i18n/src/lang/newTranslations.json",
```
Comment (esizer) [inline package.json:26]: 🤦‍♀️ Oopsies!

**Pull request merged**